tests/Serializer/SerializerTest.php

```diff
@@ -96,6 +99,47 @@ public function testNull(bool $serializeAllObjects): void
         $this->assertNull($result);
     }
 
+    public function testRegisteredObjectSerializers(): void
+    {
+        $serializer = $this->createSerializer(new Options([
+            'class_serializers' => [
+                StubObject::class => $customSerializerCallback = static function (StubObject $object): array {
+                    return [
+                        'purpose' => $object->getPurpose(),
+                    ];
+                },
+            ],
+        ]));
+
+        $object = new StubObject();
```

Comment:
Instead of a real object cannot we simply mock the interface and its methods as always?

Reply:
This has no interface, it needs a real object to be able to register the class as a serializable, the StubObject class is just a class with a public method.

Reply:
WDYT about using an anonymous class?

Reply:
Thought this was more understandable, and thought it wouldn't work. It does.

Changed.